server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -4810,9 +4810,12 @@ protected String validateUserData(String userData, HTTPMethod httpmethod) {
     }
 
     @Override
-    @ActionEvent(eventType = EventTypes.EVENT_VM_CREATE, eventDescription = "starting Vm", async = true)
+    @ActionEvent(eventType = EventTypes.EVENT_VM_CREATE, eventDescription = "deploying Vm", async = true)
```

Comment:
@shwstppr , it seems to me this should not yield an event for `EVENT_VM_CREATE` but for EVENT_VM_START`. The `EVENT_VM_CREATE` should already have been yielded on `_userVmService.createVirtualMachine(this)`.

Reply:
@DaanHoogland As of now, when deploying a new VM, CloudStack doesn't publishes an `EVENT_VM_START` event and `EVENT_VM_CREATE` is not published during `_userVmService.createVirtualMachine(this)`.
I feel this is apt behaviour because `_userVmService.createVirtualMachine(this)` will just create a DB record and actual deployment could fail in many scenarios.
Also, for a BaseAsyncCmd event with `Scheduled` state is published only when `execute` method of the API command class is executed, https://github.com/apache/cloudstack/blob/main/server/src/main/java/com/cloud/api/ApiServer.java#L732-L734. Therefore, even if we refactor `EVENT_VM_CREATE` publishing on userVmService.createVirtualMachine and `EVENT_VM_START` on `userVmService.startVirtualMachine`, without major refactor ApiServer order of events would look like,
```
EVENT_VM_CREATE Started
EVENT_VM_CREATE Completed
EVENT_VM_CREATE Scheduled
EVENT_VM_STARTED Started
EVENT_VM_STARTED Completed
```

Reply:
That first point you make makes sense, but what is the order of events now?

Reply:
I think the 'deploy' would make sense for both newly deployed and started.

Reply:
@DaanHoogland updated PR desc with tests cc @rohityadavcloud